test/preprocessing/test_preprocessing.py

```diff
@@ -42,6 +45,51 @@ def test_verify_pixel_variation(self):
         self.assertGreater(np.amax(self.raw), np.amin(self.raw))
         self.assertGreater(np.amax(self.comp), np.amin(self.comp))
 
+    def test_truncate_filenames(self):
+        """
+        Test that truncate_filenames does not crash
+
+        This test creates a directory structure with an image in it,
+        truncates filenames inside it, and basically checks that it
+        does not crash.
+
+        Ideally, this test would create a file with too long filename,
+        run truncate_filenames, and check that the filename now is
+        sufficiently short. However, it was hard to create a file wih
+        long filename, so I did not spend more time on this.
+        """
+
+        # Select a unique directory name for a new directory
+        outer_dir_name = 'temp_dir'
+        outer_dir_path = self.test_dir.joinpath(outer_dir_name)
+        while outer_dir_path.is_dir():
```

Comment:
This directory is not created, so this will never exist. I.e. the `while` statement will always be `False`

Reply:
The test now creates a duplicate folder which forces this while loop to be entered during test.

Reply:
Refactored this function, and added creation of a duplicate folder, so that this while loop indeed is being entered during test.